dali/operators/generic/reshape.cc

```diff
@@ -370,6 +413,23 @@ void Reshape<GPUBackend>::RunImpl(DeviceWorkspace &ws) {
   out.SetLayout(layout);
 }
 
+template <typename Backend>
+void Reshape<Backend>::CheckSrcDims(const Workspace &ws) {
+  if (src_dims_.empty()) {
+    return;
+  }
+
+  const auto &in = ws.template InputRef<Backend>(0);
+  const auto &input_shape = in.shape();
+  const int ndim = input_shape.sample_dim();
+  for (size_t d = 0; d < src_dims_.size(); d++) {
+    DALI_ENFORCE(-1 <= src_dims_[d] && src_dims_[d] < ndim,
+      make_string(OpName(), ": Out of bounds ``src_dims`` index. The indices in ``src_dims``"
+      " should be either a valid dimension index (range 0..ndim-1) or -1 to insert a new dimension."
+      " Got: src_dims[", d, "]=", src_dims_[d], ", ndim=", ndim));
+  }
```

Comment:
```suggestion
        make_string(OpName(), ": ``src_dims[", i, "]`` == ", src_dims[i], " is out of bounds.\n"
        "The indices in ``src_dims`` should be either valid dimension indices in range [0..", ndim-1, "] "
        "or -1 to insert a new dimension.")
  }
```
There's also the case of 0D input (in which case the only valid src_dim is -1), but I don't think it deserves a dedicated message.